pkg/machineproviders/providers/openshift/machine/v1beta1/failuredomain/failuredomain.go

```diff
@@ -276,3 +313,24 @@ func gcpFailureDomainToString(fd machinev1.GCPFailureDomain) string {
 
 	return unknownFailureDomain
 }
+
+// openstackFailureDomainToString converts the OpenStackFailureDomain into a string.
+func openstackFailureDomainToString(fd machinev1.OpenStackFailureDomain) string {
+	// ComputeAvailabilityZone only
+	if fd.ComputeAvailabilityZone != "" && fd.StorageAvailabilityZone == "" {
+		return fmt.Sprintf("OpenStackFailureDomain{ComputeAvailabilityZone:%s}", fd.ComputeAvailabilityZone)
+	}
+
+	// StorageAvailabilityZone only
+	if fd.ComputeAvailabilityZone == "" && fd.StorageAvailabilityZone != "" {
+		return fmt.Sprintf("OpenStackFailureDomain{StorageAvailabilityZone:%s}", fd.StorageAvailabilityZone)
+	}
+
+	// ComputeAvailabilityZone and StorageAvailabilityZone
+	if fd.ComputeAvailabilityZone != "" && fd.StorageAvailabilityZone != "" {
+		return fmt.Sprintf("OpenStackFailureDomain{ComputeAvailabilityZone:%s, StorageAvailabilityZone:%s}", fd.ComputeAvailabilityZone, fd.StorageAvailabilityZone)
+	}
+
+	// If a machine has no explicit zone, then the corresponding CPMS failure domain should have no explicit zone.
+	return "OpenStackFailureDomain{}"
```

Comment:
Would return `unknownFailureDomain` as other we do for the other supported platforms.

```suggestion
	return unknownFailureDomain
```

Reply:
The empty string signals OpenStack to pick the default zone.

I would even go as far as stringifying to `OpenStackFailureDomain{ComputeAvailabilityZone:"",StorageAvailabilityZone:""}` (by always using the full `Sprintf` call, perhaps with `%q`), but I suppose it's unnecessary.

Reply:
Why would a user not be explicit about which zone, rather than using the default zone, if they have gone to the trouble of specifying failure domains